cpp/src/arrow/testing/util.h

```diff
@@ -181,4 +181,11 @@ UnionTypeFactories() {
 // Windows.
 ARROW_TESTING_EXPORT int GetListenPort();
 
+namespace test {
```

Comment:
IMHO we shouldn't introduce this namespace, or we should put all functions in this file in it.